models/montecarlo.py

```diff
@@ -0,0 +1,94 @@
+from models.model import Model
+from datetime import datetime
+from environment import Environment, preprocess_data
+import numpy as np
+import pandas as pd
+import random
+import matplotlib.pyplot as plt
+
+'''
+A monte carlo model with epsilon greedy policy.
+
+- Episode: A complete run of the model from the start to the end of the training data.
+- Epsilon: The probability of choosing a random action instead of the best action for exploring instead of exploiting.
+- State-Action Dictionary: A dictionary that stores the average reward and count of each state-action pair.
+
+'''
+
+class MonteCarlo(Model):
+    def __init__(self, cleaned_data, actions, epsilon=0.1):
+        super().__init__(state_space=None, action_space=actions)
```

Comment:
The state space should be pass to the `super().__init__()` instead of writing in the initialize policy directly.